src/DataProtection/DataProtection/src/KeyManagement/KeyRingBasedDataProtector.cs

```diff
@@ -14,8 +14,233 @@
 
 namespace Microsoft.AspNetCore.DataProtection.KeyManagement
 {
+#if NETCOREAPP
```

Comment:
To start I just forked the existing Protect code to call into the Span version of code when available, didn't look to reuse any code between codepaths for now.